java/pgv-java-stub/src/main/java/io/envoyproxy/pgv/StringValidation.java

```diff
@@ -78,17 +84,27 @@ public static void suffix(String field, String value, String suffix) throws Vali
         }
     }
 
-    private static final Pattern emailWithDisplayName = Pattern.compile(".*<(.*)>");
-    public static void email(String field, String value) throws ValidationException {
-        EmailValidator emailValidator = EmailValidator.getInstance(true, true);
+    public static void email(final String field, String value) throws ValidationException {
+        if (value.charAt(value.length() - 1) == '>') {
```

Comment:
Please add a comment explaining what pattern this state machine matches. Also, does the matching logic need additional tests?

Reply:
I think that the existing tests already cover the intended logic but we can add more negative ones.

Reply:
There was actually a bug so adding tests was totally worth it. 👍